src/Module/WebFinger.php

```diff
@@ -0,0 +1,235 @@
+		if ($app->getURLPath()) {
+			$addr .= '/' . $app->getURLPath();
+		}
+
+		if ($mode == 'xml') {
+			self::printXML($addr, $alias, $profileURL, $app->getBaseURL(), $user);
+		} else {
+			self::printJSON($addr, $alias, $profileURL, $app->getBaseURL(), $user);
+		}
+	}
+
+	private static function printJSON($uri, $alias, $orofileURL, $baseURL, $user)
+	{
+		$salmon_key = Salmon::salmonKey($user['spubkey']);
+
+		header('Access-Control-Allow-Origin: *');
+		header("Content-type: application/json; charset=utf-8");
+
+		$json = ['subject' => $uri,
+				 'aliases' => [
+					 $alias,
+					 $orofileURL,
+				 ],
+				 'links'   => [
+					 [
+						 'rel'  => NAMESPACE_DFRN,
+						 'href' => $orofileURL,
+					 ],
+					 [
+						 'rel'  => NAMESPACE_FEED,
+						 'type' => 'application/atom+xml',
+						 'href' => $baseURL . '/dfrn_poll/' . $user['nickname'],
+					 ],
+					 [
+						 'rel'  => 'http://webfinger.net/rel/profile-page',
+						 'type' => 'text/html',
+						 'href' => $orofileURL,
+					 ],
+					 [
+						 'rel'  => 'self',
+						 'type' => 'application/activity+json',
+						 'href' => $orofileURL,
+					 ],
+					 [
+						 'rel'  => 'http://microformats.org/profile/hcard',
+						 'type' => 'text/html',
+						 'href' => $baseURL . '/hcard/' . $user['nickname'],
+					 ],
+					 [
+						 'rel'  => NAMESPACE_POCO,
+						 'href' => $baseURL . '/poco/' . $user['nickname'],
+					 ],
+					 [
+						 'rel'  => 'http://webfinger.net/rel/avatar',
+						 'type' => 'image/jpeg',
+						 'href' => $baseURL . '/photo/profile/' . $user['uid'] . '.jpg',
+					 ],
+					 [
+						 'rel'  => 'http://joindiaspora.com/seed_location',
+						 'type' => 'text/html',
+						 'href' => $baseURL,
+					 ],
+					 [
+						 'rel'  => 'salmon',
+						 'href' => $baseURL . '/salmon/' . $user['nickname'],
+					 ],
+					 [
+						 'rel'  => 'http://salmon-protocol.org/ns/salmon-replies',
+						 'href' => $baseURL . '/salmon/' . $user['nickname'],
+					 ],
+					 [
+						 'rel'  => 'http://salmon-protocol.org/ns/salmon-mention',
+						 'href' => $baseURL . '/salmon/' . $user['nickname'] . '/mention',
+					 ],
+					 [
+						 'rel'      => 'http://ostatus.org/schema/1.0/subscribe',
+						 'template' => $baseURL . '/follow?url={uri}',
+					 ],
+					 [
+						 'rel'  => 'magic-public-key',
+						 'href' => 'data:application/magic-public-key,' . $salmon_key,
+					 ],
+					 [
+						 'rel'  => 'http://purl.org/openwebauth/v1',
+						 'type' => 'application/x-zot+json',
+						 'href' => $baseURL . '/owa',
+					 ],
+				 ],
+		];
+
+		echo json_encode($json);
+		exit();
+	}
+
+	private static function printXML($uri, $alias, $profileURL, $baseURL, $user)
+	{
+		$salmon_key = Salmon::salmonKey($user['spubkey']);
+
+		header('Access-Control-Allow-Origin: *');
+		header("Content-type: text/xml");
+
+		$tpl = Renderer::getMarkupTemplate('xrd_person.tpl');
+
+		$o = Renderer::replaceMacros($tpl, [
+				'$nick'        => $user['nickname'],
```

Comment:
Formatting: This should be only one tab right from `$o`.

Reply:
But this one is embedded in a function call. So one tab because of the function and a second because of the array. Because otherwise the closing `]` would be at the same line as the closing `)`.

Reply:
Ah, I added `]` at the same line as `)` and now it's working :-)

Reply:
Here you go!